### DISCUSSION TITLE
Error converting FHIR resource from XML to JSON

### DISCUSSION THREAD

**Discussion by iamalonso** [nextgenhealthcare/connect#4604] [General]
Hello, I'm using Mirth 3.11 and I would like to feed a HAPI FHIR server with an OBR HL7v2.6 message.

I have 2 channels, 1 to receive the ORU-R01 that Write the HL7 into a file and then Send the HL7 message to the second channels that create a FHIR resource to be send it to HAPI FHIR.

The Destination shows the following error:

> Error converting FHIR resource from XML to JSON [DataFormatException: DataFormatException at [[row,col {unknown-source}]: [1,1]]: Unknown resource name "HL7Message" (this name is not known in FHIR version "R4")]

And the response is:

> ERROR: Error converting message or evaluating filter/transformer

I don't know what I'm doing wrong, and I've been trying different combinations of inbound/outbound data types without success.

My inbound/outbound data types configurations are:

![image](https://user-images.githubusercontent.com/13266289/122597897-0e358080-d03a-11eb-9f2f-de40ad396452.png)

![image](https://user-images.githubusercontent.com/13266289/122597915-1392cb00-d03a-11eb-9960-ece5dce29ddd.png)

**Comment by tom08zehn** [comment]
I didn’t use FHIR so far but I see in your settings that in the 2nd channel the inbound message is of type HL7 (is converted to XML with root node name "HL7Message", the one in your error message) and you want FHIR as outbound which requires is a different payload.

You need to do the conversion, Mirth doesn‘t do it for you automatically.

When I remember my early steps with Mirth I’ve had the same expectation as you: HL7 inbound to JSON outbound - won’t work unless you convert in the transformer on your own.

Also see:

HL7 to FHIR examples: http://www.mirthcorp.com/community/wiki/plugins/servlet/mobile#content/view/36504968

And for ORU: https://forums.mirthproject.io/forum/mirth-connect/support/17913-hl7-to-fhir-observations

For the XML to JSON conversion: http://javadocs.mirthcorp.com/connect/3.6.0/user-api/index.html?com/mirth/connect/userutil/XmlUtil.html

Example:
```js
var jsonString = XmlUtil.toJson(xmlString);
```

**Reply by iamalonso** [reply]
Hello @tom08zehn thanks for your reply. Actually I'm doing the HL7v2.6 to FHIR mapping using the FHIR plugin (NextGen Healthcare) for Mirth Connect - See the image below. So I don't understand in which step I have to do the conversion did you mentioned before. I thought that the conversion was done automatically by the plugin.

![image](https://user-images.githubusercontent.com/13266289/122649718-2c17e980-d0fd-11eb-9426-2a06c2b7411c.png)
